QA/PoroDevTest/src/main/java/common/ui_setup/pom_setup/PoroDevPom/RegistrationPage.java

```diff
@@ -1,89 +1,132 @@
+    @FindBy(xpath = "//*[@id=\"registerForm\"]/div[2]/span[2]")
+    public WebElement we_lastName_requiredMsg;
+    @FindBy(xpath = "//*[@id=\"registerForm\"]/div[3]/span[2]")
+    public WebElement we_email_requiredMsg;
+    @FindBy(xpath = "//body/div[@id='root']/div[1]/div[1]/form[1]/div[4]/span[3]")
+    public WebElement we_password_requiredMsg;
+    @FindBy(xpath = "//body/div[@id='root']/div[1]/div[1]/form[1]/div[5]/span[3]")
+    public WebElement we_confirmPass_requiredMsg;
+    @FindBy(xpath = "//*[@id=\"registerForm\"]/div[6]/span[2]")
+    public WebElement we_department_requiredMsg;
+    @FindBy(xpath = "//*[@id=\"registerForm\"]/div[7]/span[2]")
+    public WebElement we_position_requiredMsg;
+    @FindBy(className = "sc-hHLeRK")
+    List<WebElement> we_errorFields;
 
     //SUCCESSFUL MESSAGE SECTION
     @FindBy(xpath = "//span[contains(text(),'Successful registration')]")
-    public WebElement successful_registrationMsg;
+    public WebElement we_successful_registrationMsg;
 
-    public void enterFirstName(String firstName) {
-        elementControl.sendText(firstName_field, firstName);
-    }
-    public void enterLastName(String lastName) {
-        elementControl.sendText(lastName_field, lastName);
-    }
-    public void enterEmail(String email) {
-        elementControl.sendText(email_field, email);
-    }
-    public void enterPassword(String pass) {
-        elementControl.sendText(password_field, pass);
-    }
-    public void enterDepartment(String department) {
-        elementControl.sendText(department_field, department);
+    public void registerUser(String firstName, String lastName, String email, String password, String confirmPass, String department, String position) throws InterruptedException {
+
+
+        BasePage.clickElement(we_createNewAccount_button);
+        BasePage.sendText(we_firstName_inputField, firstName);
+        BasePage.sendText(we_lastName_inputField, lastName);
+        BasePage.sendText(we_email_inputField, email);
+        BasePage.sendText(we_password_inputField, password);
+        BasePage.sendText(we_confirmPass_inputField, confirmPass);
+        BasePage.sendText(we_department_inputField, department);
+        BasePage.sendText(we_position_inputField, position);
+        BasePage.clickElement(we_register_button);
     }
-    public void enterPosition(String position) {
-        elementControl.sendText(position_field, position);
+
+    public void explicitWait(WebElement element) {
+        wait = new WebDriverWait(driver, Duration.ofSeconds(SetupConstants.ELEMENT_DETECTION_TIMEOUT));
+        wait.until(ExpectedConditions.visibilityOf(element));
     }
-    public void clickRegisterButton() {
-        elementControl.clickElement(register_button);
+
+    //ASSERT METHODS
+    public void assert_user_registration(WebElement element, String expectedResults) {
+        System.out.println("error: " + BasePage.getTextFromElement(element));
```

Comment:
In future commits, each of sout should be replaces with logger calls

Reply:
Got it